core/src/lib.rs

```diff
@@ -12,6 +12,7 @@ extern crate alloc;
 
 pub mod cursor;
 pub mod multi_proof;
+pub mod multi_proof_verification;
```

Comment:
style: I'd prefer either:
  - a single `multi_proof` module with two child modules (prove and verify), or
  - a single `multi_proof` module with all the code within it

instead of this.